### ISSUE TITLE
Is it really necesary to download the whole chat history?

### ISSUE THREAD

**Issue by DuendeInexistente** [EionRobb/purple-discord#194]
As the title says, is it? I find it honestly annoying that, when launching this plugin for the first time in a while, the first thing is having to download literal months worth of chatlogs on every room I set to auto connect. Even disregarding the delay this adds to being able to actually chat, having all the pings from those mentions is just plain distracting.

**Comment by EionRobb** [comment]
Ugh history is always so painful 🙂 

Help me out here, what do you think is the right behaviour? When you disconnect and reconnect to the account, you want to fetch any messages you missed while you were offline. And in a perfect world, the pidgin history would match the servers history. But at what point do we break that and say "I'm ok with gaps in pidgins history logging"?

**Comment by DuendeInexistente** [comment]
Is it possible to first download say, the last day of history, and then the rest in the background? I have the feeling that with how Pidgin's history logging works it may not like downloading it non-linearly.

I *do* like having this history at hand locally, mind you. Maybe the best thing until someone thinks of a better solution would be to make it a choice between downloading all of it, the last day/s, or none.

**Comment by EionRobb** [comment]
The default html logger just stores it as it's given, so would look messed up when viewing the logs. The sqlite logger will reorganize the messages based on timestamp, but most people don't use that one. 

The setting idea isn't a bad one 🙂 what should the default be, in your opinion?

**Comment by DuendeInexistente** [comment]
Last day sounds like a decent compromise.
Besides that maybe options for the last hour, week, month and all. I imagine people with huge chatrooms would only ever want the last hour.

**Comment by alexolog** [comment]
I would suggest using the current behaviour as the default, to avoid surprises.

**Comment by bobbybee** [comment]
> I would suggest using the current behaviour as the default, to avoid surprises.

^^ This is objectively correct from a stability standpoint, absent a
strong compelling reason to the contrary, given we already have a
nontrivial number of users.

**Comment by alexolog** [comment]
The history retrieval doesn't work for me at all.

Scenario #1:
Pidgin is closed.  I have a private conversation with a buddy using the official desktop client.  When I reopen Pidgin and click on the buddy, that conversation is not retrieved.

Scenario #2:
I open a Discord channel in Pidgin.  The window has no text in it and no history is retrieved.

**Comment by mattdawolf** [comment]
> The default html logger just stores it as it's given, so would look messed up when viewing the logs. The sqlite logger will reorganize the messages based on timestamp, but most people don't use that one.
> 
> The setting idea isn't a bad one slightly_smiling_face what should the default be, in your opinion?

This is good behavior to have, but I think it's stopped working. It should pull/log everything that it missed.

**Comment by GabeIggy** [comment]
The current behavior is painful because whenever you join a new channel (freshly configured Pidgin), it will download the entire history. This makes Pidgin unusable when you have a channels with a long history span (2018-2019) and it will really slow down legacy systems.

Downloading the entire history should be an option.  Give the user the freedom to specify the history limit.

**Comment by mattdawolf** [comment]
I agree. Make it an option but still have the ability. Mine has some gaps
that need to be filled from the server.
